### PR TITLE
Updated code to work with GPR

### PR DESCRIPTION
**Pull request by DmitryLegostaev**
- Fixed source add for GPR
- Made nuget push request working for GPR and a bit more readable. Looks like it could be simplified but it requires time for testing to be sure that it will work.
- API request endpoint used in "GPR small hack" is moved permanently. Added 301 status handling just to make it working. It doesn't provide any error handling or an alternative endpoint to use.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ class Action {
             let addSourceCmd;
             if (this.nugetSource.startsWith(`https://nuget.pkg.github.com/`)) {
                 this.sourceType = "GPR"
-                addSourceCmd = `dotnet nuget add source ${this.nugetSource}/index.json --name=${(this.sourceName)} --username=${this.githubUser} --password=${this.nugetKey} --store-password-in-clear-text`
+                addSourceCmd = `dotnet nuget add source ${this.nugetSource}/${this.githubUser}/index.json --name=${(this.sourceName)} --username=${this.githubUser} --password=${this.nugetKey} --store-password-in-clear-text`
```
Comment (Rebel028) [inline index.js:36]: This is a breaking change that will cause failures in existing workflows. Sorry, not going to merge it. 

Also it won't let people to use organization repos and that's the main reason why `nugetSource` parameter looks like that: it suggests that you will use string like `https://nuget.pkg.github.com/orgname` or `https://nuget.pkg.github.com/username` depending on what type of repository you have. 
UPD.It affects shared repos that are owned by other people too.

**Comment by Rebel028** [comment]
> * API request endpoint used in "GPR small hack" is moved permanently.

I don't see any problems with it. Need more info on that, what's full URL you use for your workflow? 
If it's your own repo or your org, have you changed your username or organisation name?

**Comment by Rebel028** [comment]
Closing this for now, looking forward to your reply on 301 redirect, so I can handle it properly if it's a real problem.

File: index.js (hunk 1 above)
Comment (DmitryLegostaev) [reply index.js:36]: Agreed on that, looks like my bad

I suppose that current users of this action are just passing full link with a namespace included, that's why it's working for them.
I might be passing 'https://nuget.pkg.github.com/NAMESPACE' however I've tried to pass just 'https://nuget.pkg.github.com/' and it didn't work for me
So in my case the action have tried to send a GET to 'https://nuget.pkg.github.com/index.json' and failed for me as this is an incorrect url.

**Comment by DmitryLegostaev** [comment]
> Closing this for now, looking forward to your reply on 301 redirect, so I can handle it properly if it's a real problem.

In my case it might be caused by the input link issue I've described above, so if it works for anyone who is passing a full link with a namespace to the action `https://nuget.pkg.github.com/NAMESPACE`, then it's not an issue at all, so this PR doesn't make much sense.

Thanks for your comments

**Pull request closed without merging**